test/unit/lib/PersistentCache.test.js

```diff
@@ -36,31 +33,23 @@ describe("PersistentCache", () => {
   });
 
   describe("#get", () => {
-    it("tries to read the file on the first get", async () => {
-      fakeOS.File.read = sinon.spy();
+    it("tries to fetch the file", async () => {
       await cache.get("foo");
-      assert.calledOnce(fakeOS.File.read);
+      assert.calledOnce(fakeFetch);
     });
-    it("doesnt try to read the file if it doesn't exist", async () => {
-      fakeOS.File.read = sinon.spy();
-      fakeOS.File.exists = async () => false;
+    it("doesnt try to parse file if it doesn't exist", async () => {
+      fakeFetch.throws();
       await cache.get("foo");
-      assert.notCalled(fakeOS.File.read);
+      assert.notCalled(fakeJsonParse);
     });
-    it("doesnt try to read the file if it was already loaded", async () => {
-      fakeOS.File.read = sinon.spy();
+    it("doesnt try to fetch the file if it was already loaded", async () => {
       await cache._load();
-      fakeOS.File.read.resetHistory();
+      fakeFetch.resetHistory();
       await cache.get("foo");
-      assert.notCalled(fakeOS.File.read);
-    });
-    it("should catch and report errors", async () => {
```

Comment:
I think it would be useful to keep the `Cu.reportError` Call if fetch or json parsing fails.

Reply:
the problem was `fetch()` throws if the cache file doesn't exist yet. So, it would be spammy to report that an an error since it is expected on a cold cache. I'll break up the `fetch` from the `json` and only report json parsing errors.